src/cli/options/bgpsec.rs

```diff
@@ -0,0 +1,176 @@
+//! Managing a CA’s router keys.
+
+use std::{error, fmt, fs, io};
+use std::convert::Infallible;
+use std::str::FromStr;
+use rpki::ca::csr::BgpsecCsr;
+use rpki::crypto::{KeyIdentifier, SignatureVerificationError};
+use rpki::dep::bcder::decode::DecodeError;
+use rpki::repository::resources::Asn;
+use crate::cli::client::KrillClient;
+use crate::cli::report::Report;
+use crate::commons::api;
+use crate::commons::util::httpclient;
+use super::ca;
+
+
+//------------ Command -------------------------------------------------------
+
+#[derive(clap::Subcommand)]
+pub enum Command {
+    /// Show current BGPsec router keys
+    List(List),
+
+    /// Add a BGPsec router key
+    Add(Add),
+
+    /// Remove a BGPsec router key
+    Remove(Remove),
+}
+
+impl Command {
+    pub async fn run(self, client: &KrillClient) -> Report {
+        match self {
+            Self::List(cmd) => cmd.run(client).await.into(),
+            Self::Add(cmd) => cmd.run(client).await.into(),
+            Self::Remove(cmd) => cmd.run(client).await.into(),
+        }
+    }
+}
+
+
+//-------- List --------------------------------------------------------------
+
+#[derive(clap::Args)]
+pub struct List {
+    #[command(flatten)]
+    ca: ca::Handle,
+}
+
+impl List {
+    async fn run(
+        self, client: &KrillClient
+    ) -> Result<api::BgpSecCsrInfoList, httpclient::Error> {
+        client.bgpsec_list(&self.ca.ca).await
+    }
+}
+
+
+//------------ Add ----------------------------------------------------------
+
+#[derive(clap::Parser)]
+pub struct Add {
+    #[command(flatten)]
+    ca: ca::Handle,
+
+    /// The ASN to authorize the router key for
+    #[arg(long, short, value_name = "ASN")]
+    asn: Asn,
+
+    /// Path to the DER-encoded certificate signing request.
+    #[arg(long, value_name = "path")]
+    csr: BgpsecCsrFile,
+}
+
+impl Add {
+    async fn run(
+        self, client: &KrillClient
+    ) -> Result<api::Success, httpclient::Error> {
+        client.bgpsec_update(
+            &self.ca.ca,
+            api::BgpSecDefinitionUpdates::new(
+                vec![api::BgpSecDefinition::new(self.asn, self.csr.0)],
+                vec![]
+            )
+        ).await
+    }
+}
+
+
+//------------ Remove --------------------------------------------------------
+
+#[derive(clap::Parser)]
+pub struct Remove {
+    #[command(flatten)]
+    ca: ca::Handle,
+
+    /// The ASN of router key to be removed
+    #[arg(long, short, value_name = "ASN")]
+    asn: Asn,
+
+    /// The hex encoded key identifier of the router key
+    #[arg(long, value_name = "key identifier")]
```

Comment:
Nit: Inconsistent case: "ASN" starts with a capital letter while "key identifier" does not.

Reply:
The value names are intended to be in regular case, i.e., starting small. But ASN is an abbreviation and hence in capitals.